### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by WinstonHall**
# Summary

Please include a summary of changes

-

## Type of change

Select all that apply (add an x inside brackets)

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

# Checklist:

-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: igdbHelper.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  accessToken = response.data.access_token;
+  tokenExperationTime = response.data.expires_in + currentTime;
+};
+
+const reusableAxiosPost = async (routeName) => {
+  await getAccessToken();
+
+  const config = newReleasesPostConfig();
+  const data = postData[routeName]();
+  return await axios.post("https://api.igdb.com/v4/games", data, config);
```
Comment (Raymond-Cox) [inline igdbHelper.js:26]: I don't think `await` does anything when you're returning a function that returns a promise. Should be able to just
```
return axios.post("https://api.igdb.com/v4/games", data, config);
```

File: igdbHelper.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+module.exports = reusableAxiosPost;
+
+//AXIOS CONFIG
+//axios posts default config
+const newReleasesPostConfig = () => {
```
Comment (Raymond-Cox) [inline igdbHelper.js:33]: The function naming is weird here. The same config is used for any axios.post call to that api.

File: igdbHelper.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  if (accessToken && currentTime < tokenExperationTime) return;
+
+  const url = `https://id.twitch.tv/oauth2/token?client_id=${CLIENT_ID}&client_secret=${CLIENT_SECRET}&grant_type=client_credentials`;
+  const response = await axios.post(url);
+
+  accessToken = response.data.access_token;
+  tokenExperationTime = response.data.expires_in + currentTime;
+};
+
+const reusableAxiosPost = async (routeName) => {
```
Comment (Raymond-Cox) [inline igdbHelper.js:21]: I think we can go with a more generic name here. Maybe just `IDGB_Post`, or just `Post`.

File: igdbHelper.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      Accept: "application/json",
+      "Client-ID": `${CLIENT_ID}`,
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "text/plain",
+    },
+  };
+};
+
+//AXIOS DATA OBJECTS
+const postData = {
```
Comment (Raymond-Cox) [inline igdbHelper.js:45]: I like the idea here, but I think a single function approach with a switch statement might scale better.

```
function formatData(dataType) {
    const currentTime = new Date().getTime();
    const startingTime = currentTime - 604800; //current time - a week in unix
    
    // Now return the 3 different data strings from a switch statement
    switch(dataType) {
      case 'new':
        return `fields release_dates.platform.*, release_dates.human, name; where release_dates.date > ${startingTime} & release_dates.date <= ${currentTime};`;
      case 'upcoming':
         ... etc
    }
}
```

**Review by Raymond-Cox — CHANGES_REQUESTED** [review]
Some humble cleanup and renaming

File: igdbHelper.js (hunk 2 above)
Comment (WinstonHall) [reply igdbHelper.js:33]: I meant to change that to match the config. I forgot it was set that way, but the goal was to put something in place incase we had to change any headers. Just got left behind and forgotten in the mess.

**Review by Raymond-Cox — APPROVED** [review]
Nice job. We'll improve test coverage next PR

**Pull request merged**